server/container/guice/guice-common/src/main/java/org/apache/james/utils/InMemoryMailRepositoryStore.java

```diff
@@ -92,15 +115,15 @@ public void init() throws Exception {
     }
 
     @Override
-    public MailRepository select(String destination) throws MailRepositoryStoreException {
-        MailRepository mailRepository = destinationToRepositoryAssociations.get(destination);
+    public MailRepository select(String destinationUrl) throws MailRepositoryStoreException {
+        Destination destination = Destination.fromUrl(destinationUrl);
+        MailRepository mailRepository = destinationToRepositoryAssociations.get(destination.url);
         if (mailRepository != null) {
             return mailRepository;
         }
-        String protocol = retrieveProtocol(destination);
-        mailRepository = retrieveMailRepository(protocol);
-        mailRepository = initialiseNewRepository(mailRepository, createRepositoryCombinedConfig(destination, protocol));
-        destinationToRepositoryAssociations.putIfAbsent(destination, mailRepository);
+        mailRepository = retrieveMailRepository(destination);
```

Comment:
Maybe we can add one commit to avoid variable reallocation and name that method part `createMailRepository`